pkg/tuf/client.go

```diff
@@ -0,0 +1,618 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tuf
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/theupdateframework/go-tuf/client"
+	tuf_leveldbstore "github.com/theupdateframework/go-tuf/client/leveldbstore"
+	"github.com/theupdateframework/go-tuf/data"
+	"github.com/theupdateframework/go-tuf/util"
+)
+
+const (
+	DefaultRemoteRoot = "sigstore-tuf-root"
```

Comment:
The GCS dependency wasn't dropped yet, so can we punt this task to later?

Reply:
That sounds great, what's the correct HTTP URL to set here, instead of this? Is it just `https://sigstore-tuf-root.storage.googleapis.com`?

And if we do that, should I remove the GCS path in `remoteFromMirror`, and make callers who want to use GCS provide their own TUF client somehow? (Also fine making this work for a future PR, if we want this PR to mainly be code movement with not much functional change)

Reply:
> GCS path in remoteFromMirror, and make callers who want to use GCS provide their own TUF client somehow?

Sure, maybe in the follow-up we should expose the remote as a tuf client.RemoteStore in the API for `Initialize` -- that way library callers can provide their open GCS/other remote store.